department-of-veterans-affairs/caseflow
# APPEALS-30891

Resolves [APPEALS-30891](https://jira.devops.va.gov/browse/APPEALS-30891)

# Description
Added the Conditions section to the Report Page 
I also made stub components for the individual condition stories to fill in later (30937 to 30942)

## Acceptance Criteria
- [ ] Creation of ‘Add condition’ button. 
- [ ] Creation of drop-down selection for filter type:
    - [ ] Days Waiting
    - [ ] Decision Review Type
    - [ ] Issue Type
    - [ ] Issue Disposition
    - [ ] Personnel
    - [ ] Facility
- [ ] Users can optionally add additional Conditions via ‘Add condition’ button.
    - [ ] If no additional conditions are input, no restrictions for optional filter conditions will be placed on report generation query.
    - [ ] Selecting a filter option will prevent it from being selected in another condition.
    - [ ] If a ‘Variable’ option is selected, that ‘Variable’ drop-down is unable to be changed to another ‘Variable’ type selection. It must be removed if user does not want to include that condition.
    - [ ] User cannot include both Personnel filter type and Facility filter type, they will be mutually exclusive.
- [ ] User can remove any selected additional Conditions displayed via ‘Remove condition’ link. 

## Testing Plan


1. Log in as ACBAUERVVHAH and navigate to /decision_reviews/vha/report
2. There should be a Conditions header and Add Condition button
3. Click Add condition - a dropdown should appear that allows you to select a condition. Pick Personnel.
4. Click Add condition again - make sure neither Personnel nor Facility appears. Select something.
5. Click the remove condition link under one of the conditions
6. Play around with adding and removing conditions and try to break it


# Frontend
## User Facing Changes

<img width="1147" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/17840614/d8d1780e-f7dd-455c-aadd-6a87c7413f5d">


## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
